### ISSUE TITLE
Basic Auth/TLS broken in marathonClient.AddEventsListener()

### ISSUE THREAD

**Issue by emilevauge** [gambol99/go-marathon#148]
When we create a marathon client with basic auth and/or TLS, the call to `marathonClient.AddEventsListener()` returns an error:

```
401: <html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 401 </title>
</head>
<body>
<h2>HTTP ERROR: 401</h2>
<p>Problem accessing /v2/events. Reason:
<pre>    Unauthorized</pre></p>
<hr /><a href="http://eclipse.org/jetty">Powered by Jetty:// 9.3.z-SNAPSHOT</a><hr/>
</body>
</html>
```

with TLS:

```
x509: certificate is valid for *.mydomain.net, mydomain.net, not ***.***.***.***.mydomain.net
```
Here is the code used to register listeners: https://github.com/containous/traefik/blob/master/provider/marathon.go#L67

It seems marathonClient.AddEventsListener() doesn't use TLS/basic auth client config.

**Comment by arichardet** [comment]
Any resolution on this?

**Comment by timoreimann** [comment]
@arichardet I suppose PR #142 will provide a fix for the problem once it has been merged.

**Comment by gambol99** [comment]
@timoreimann ... i think #142 fixes the auth but not TLS. I've raised a PR https://github.com/donovanhide/eventsource/pull/22 to permit overriding the client

**Comment by timoreimann** [comment]
Good point, those are indeed two different issues. Thanks!

**Comment by gambol99** [comment]
was fixed in #156

**Comment by emilevauge** [comment]
Thanks a lot @gambol99 :)
